### ISSUE TITLE
Rule request: enforce best practices when mocking

### ISSUE THREAD

**Issue by mgaitan** [astral-sh/ruff#12850]
I'd like to propose the integration of `flake8-patch` like rule/s  into `ruff` to help enforce best practices when mocking in Python tests.

In a legacy codebase, we've encountered issues where direct monkeypatching  leads to tests that do not properly clean up after themselves. This can result in flaky tests that pass or fail depending on the order of execution. For example, we had the following pattern:

```python
Shipment.get_label_info = mocker.Mock()
Shipment.get_label_info.return_value = (None, tracking_url, None)
```

This approach led to issues in teardown, as the mock was not automatically undone. We switched to using `mocker.patch.object()`:

```python
mocker.patch.object(Shipment, "get_label_info", return_value=(None, tracking_url, None))
```

This approach ensures that the patch is properly removed after the test.

The [flake8-patch](https://pypi.org/project/flake8-patch/) plugin helps identify improper mocking practices, such as direct assignment to imported objects, which can cause tests to be unreliable. Integrating similar functionality into `ruff` would allow us to catch these issues early in the development process.
